### PR TITLE
Menu refactor & Repo switcher

### PR DESCRIPTION
**Pull request by himdel**
Fixes https://issues.redhat.com/browse/AAH-485

This moves the repository select from the menu to the relevant screens, and updates the menu to a new design, with sections.

Menu change:

|**before**|**after** - expanded|**after** - collapsed|
|-|-|-|
|  |  |  |

The menu is also highlighting the active item now, based on url prefix match,
remembering open sections during navigation (but on reload, defaults to only expanding the current section),
and hiding empty sections when the user can't access any items under these.

The question mark menu is updated to use external links with the icon, and Documentation moved to menu..

Screens with the selector:

* `src/containers/search/search.tsx` (r/w, goes to `/ui/repo/:selectedRepo`)
  * `/ui/repo/:selectedRepo/` - Collections

* `src/components/headers/collection-header.tsx` (r/o)
  * `/ui/repo/:selectedRepo/:namespace/:collection` - Collection detail, tab Details
  * `/ui/repo/:selectedRepo/:namespace/:collection/docs` - Collection detail, tab Documentation
  * `/ui/repo/:selectedRepo/:namespace/:collection/content` - Collection detail, tab Contents
  * `/ui/repo/:selectedRepo/:namespace/:collection/import-log` - Collection detail, tab Import log

* `src/containers/namespace-detail/namespace-detail.tsx` via `src/components/headers/partner-header.tsx` (r/w, goes to TODO)
  * `/ui/repo/:selectedRepo/:namespace` - Collections under a namespace
  * `/ui/repo/:selectedRepo/my-namespaces/:namespace` - My Collections under a namespace
  * `/ui/my-namespaces/:namespace` - My Collections under a namespace (alias)

(+ artefact: created a [list of screens](https://github.com/himdel/ansible-hub-ui/wiki/Screens))

### CODE REVIEW

**Comment by sbuenafe-rh** [comment]
@himdel @newswangerd I would keep just Documentation (no expansion) on the left nav and move **Customer Support**, **Training**, and **About** to the ? icon in the masthead

**Comment by himdel** [comment]
Updated :)
Documentation is now a single menu item, with the rest remaining in the top dropdown,
menu remembers expanded/collapsed sections until a full page reload,
menu autoexpands the active section on load, but not after,
the repo selector should be correctly sized now,
read only on all the collection detail screens, read-write on the collections screen and namespace detail screens,
and it can switch to namespaceByRepo/myCollectionsByRepo based on the current route.

Ready for review :)

**Comment by sbuenafe-rh** [comment]
@himdel I would move the breadcrumbs underneath the context selector. Within the context selector I would prepend **Filter by repository: Red Hat Certified** so the user knows what they are filtering by. Other than those 2 things, it looks good. 

@newswangerd Default repository is Red Hat Certified, correct?

**Comment by newswangerd** [comment]
@sbuenafe-rh default is published

**Comment by himdel** [comment]
@sbuenafe-rh  Thanks, updated :)..

**Review by newswangerd — APPROVED** [review]

**Comment by himdel** [comment]
Breaks cypres..
    ✖  3 of 5 failed (60%)                      00:42       13        2        6        1        4  

Updated menu helpers to understand the new structure..
    ✔  All specs passed!                        01:14       13       12        -        1        -

File: test/cypress/support/commands.js [hunk 1]
```diff
@@ -39,9 +39,21 @@ Cypress.Commands.add('containsnear', {}, (...args) => {
...
+Cypress.Commands.add('menuMissing', {}, (name) => {
+    const last = name.split(' > ').pop();
+    return cy.contains('#page-sidebar a', last).should('not.exist');
+});
+
+Cypress.Commands.add('menuGo', {}, (name) => {
+    const last = name.split(' > ').pop();
+    return cy.contains('#page-sidebar a', last).click({ force: true });
+});
+
```
Comment (hendersonreed) [inline test/cypress/support/commands.js:56]: Great thinking on this refactor! Adding these commands simplifies things quite a bit.

**Review by hendersonreed — APPROVED** [review]
This looks great from the test side (that's all I feel like I have the background to review atm :)

**Pull request merged**